patch_recurrence/steps.py

```diff
@@ -33,16 +34,59 @@ def generate_patches(scaled_imgs, constants):
                     raw_patch=raw_patch,
                     patch_size=patch_size,
                 )
-                if patch.std_dev > std_dev_threshold:
+                if all_patches_required or patch.std_dev > std_dev_threshold:
                     patch.store(sc, [i, j])
                     img_patches.append(patch)
         patches.append(img_patches)
     return patches
 
 
+'''
+Applying Gaussian filter to smoothen std deviations of all patches
+'''
+def smoothen(scaled_imgs, patches, constants):
+    patch_size = constants.PATCH_SIZE
+    img = scaled_imgs[0]
+    patch = patches[0]
+
+    length_sd_array = img.shape[0] - patch_size
+    width_sd_array = img.shape[1] - patch_size
+    patch = np.reshape(patch, [length_sd_array, width_sd_array])
```

Comment:
This will internally convert `patch` into a `numpy` array of objects. This is how I would do it -
````
std_database = np.reshape(map(lambda x: x.std_dev, patch), [length_sd_array, width_sd_array])
blur = np.reshape(cv2.GaussianBlur(std_database, (7, 7), sigmaX=6, sigmaY=6), [-1])
map(lambda (i, x): setattr(x, 'std_dev', blur[i]), enumerate(patch))
````